python-flask/python-flask-example.py

```diff
@@ -13,20 +13,18 @@ def log_payload(path):
         return 'Content type is not JSON', 415
 
     # Ensure that payload is a dictionary
-    payload = request.json
-    if not isinstance(payload, dict):
+    data = request.json
+    if not isinstance(data, dict):
         return 'Payload is not JSON', 500
 
     # Get message data
-    timestamp = datetime.datetime.now()
-    message = payload.get('message',{})
-    if message:
+    device_id = data.get('deviceId', {})
+    if device_id:
         # Get device ID
-        id = message.get('attributes',{}).get('deviceId','')
-        # Get base64 encoded data payload
-        data = base64.b64decode(message.get('data',''))
-        print('----- Got message from {} ({}) -------'.format(id, timestamp))
-        pprint.pprint(json.loads(data))
+        payload = data.get('payload', {})
+        timestamp = data.get('timestamp', datetime.datetime.now())
```

Comment:
```suggestion
        timestamp = data.get('timestamp', '(no timestamp provided)')
```
A timestamp is always provided, but this makes it more clear in the very unlikely case that it isn't.

Reply:
true, changed it